rest_framework/fields.py

```diff
@@ -71,7 +71,11 @@ def get_attribute(instance, attrs):
         except ObjectDoesNotExist:
             return None
         if is_simple_callable(instance):
-            instance = instance()
+            try:
+                instance = instance()
+            except (AttributeError, KeyError) as exc:
+                raise CallableAttributeExceptionWrapper(exc)
```

Comment:
Sounds good -- I just got there myself after looking for the other places that call the `get_attribute` helper.